### PR TITLE
[S1APTest] Add continuous random attach test

### PR DESCRIPTION
**Pull request by talkhasib**
Signed-off-by: Tariq Al-Khasib <talkhasib@fb.com>
## Summary

Add a test that simulates random attach/detach.
You can control the number of UEs, attach rate, and the duration before the UE detaches

## Test Plan

Tested on Vagrant VMs

### CODE REVIEW

**Review by ssanadhya — APPROVED** [review]

**Comment by ssanadhya** [comment]
Do check the static code analysis CI checks at https://github.com/magma/magma/pull/4747/checks?check_run_id=1826327586 .

File: lte/gateway/python/integ_tests/s1aptests/test_continuous_random_attach.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_CTX_REL_IND.value:
+            m = msg.cast(s1ap_types.FwNbUeCtxRelInd_t)
+            return m.ueId
+        elif msg.msg_type == s1ap_types.tfwCmd.UE_IDENTITY_REQ_IND.value:
+            m = msg.cast(s1ap_types.ueIdentityReqInd_t)
+            return m.ue_Id
+
+        return 0
+
+    def test_continuous_random_attach(self):
```
Comment (ulaskozat) [inline lte/gateway/python/integ_tests/s1aptests/test_continuous_random_attach.py:149]: is the test intended to run forever? can you add a test duration?

File: lte/gateway/python/integ_tests/s1aptests/test_continuous_random_attach.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    def handle_detach_timer(self, req):
+        print("Detaching ", req.ue_id)
+        self.send_ue_detach(req.ue_id)
+        attachTime = random.uniform(self.attach_delay_t0, self.attach_delay_t1)
+        attachTimer = threading.Timer(attachTime, self.handle_attach_timer, args=(req,))
+        attachTimer.start()
+
+    def handle_attach_timer(self, req):
+        print("Attaching ", req.ue_id)
+        attachDuration = random.uniform(self.attach_duration_t0, self.attach_duration_t1)
```
Comment (ulaskozat) [inline lte/gateway/python/integ_tests/s1aptests/test_continuous_random_attach.py]: can we make this configurable to use different distributions?

File: lte/gateway/python/integ_tests/s1aptests/test_continuous_random_attach.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+            x = threading.Thread(target=self.ue_thread_function, args=(req, q))
+            threads.append(x)
+            queues.append(q)
+            x.start()
+
+        while True:
+            response = self._s1ap_wrapper.s1_util.get_response()
+            ueId = self.get_ue_id(response)
+            if ueId > 0:
+                queues[ueId-1].put(response)
```
Comment (ulaskozat) [inline lte/gateway/python/integ_tests/s1aptests/test_continuous_random_attach.py]: are we using this queue for any action? it looks like we are only storing up to 10 responses per UE or am I misreading it?

**Review by ulaskozat — APPROVED** [review]
some nits left. this test case can serve as a good basis for a fuzz test.

File: lte/gateway/python/integ_tests/s1aptests/test_continuous_random_attach.py (hunk 1 above)
Comment (talkhasib) [reply lte/gateway/python/integ_tests/s1aptests/test_continuous_random_attach.py:149]: That was the use case for me so far. But, makes sense to add a test duration. Let me do that.

File: lte/gateway/python/integ_tests/s1aptests/test_continuous_random_attach.py (hunk 3 above)
Comment (talkhasib) [reply lte/gateway/python/integ_tests/s1aptests/test_continuous_random_attach.py]: Each thread is a consumer of its corresponding queue. I dont see this queue filling up with more than 10 messages per UE

**Comment by talkhasib** [comment]
Gave it a major face lift. Worth a second review

**Pull request merged**